pints/_abc/__init__.py

```diff
@@ -0,0 +1,345 @@
+        """
+        Enables or disables progress logging to screen.
+        """
+        self._log_to_screen = True if enabled else False
+
+    def max_iterations(self):
+        """
+        Returns the maximum iterations if this stopping criterion is set, or
+        ``None`` if it is not. See :meth:`set_max_iterations()`.
+        """
+        return self._max_iterations
+
+    def nr_samples(self):
+        """
+        Returns the target number of samples to obtain in the estimated
+        posterior.
+        """
+        return self._nr_samples
+
+    def parallel(self):
+        """
+        Returns the number of parallel worker processes this routine will be
+        run on, or ``False`` if parallelisation is disabled.
+        """
+        return self._n_workers if self._parallel else False
+
+    def run(self):
+        """
+        Runs the ABC sampler.
+        """
+        if self._max_iterations is None:
+            raise ValueError("At least one stopping criterion must be set.")
+
+        # Iteration and evaluation counting
+        iteration = 0
+        evaluations = 0
+        accepted_count = 0
+
+        # Choose method to evaluate
+        f = self._error_measure
+
+        # Create evaluator
+        if self._parallel:
+            n_workers = self._n_workers
+            evaluator = pints.ParallelEvaluator(f, n_workers=n_workers)
+        else:
+            evaluator = pints.SequentialEvaluator(f)
+
+        # Set up progress reporting
+        next_message = 0
+
+        # Start logging
+        logging = self._log_to_screen or self._log_filename
+        if logging:
+            if self._log_to_screen:
+                print('Using ' + str(self._sampler.name()))
+                if self._parallel:
+                    print('Running in parallel with ' + str(n_workers) +
+                          ' worker processess.')
+                else:
+                    print('Running in sequential mode.')
+
+            # Set up logger
+            logger = pints.Logger()
+            if not self._log_to_screen:
+                logger.set_stream(None)
+            if self._log_filename:
+                logger.set_filename(self._log_filename, csv=self._log_csv)
+
+            # Add fields to log
+            max_iter_guess = max(self._max_iterations or 0, 10000)
+            max_eval_guess = max_iter_guess
+            logger.add_counter('Iter.', max_value=max_iter_guess)
+            logger.add_counter('Eval.', max_value=max_eval_guess)
+            logger.add_float('Acceptance rate')
+            self._sampler._log_init(logger)
+            logger.add_time('Time m:s')
+
+        # Start sampling
+        timer = pints.Timer()
+        running = True
+
+        samples = []
+        # Sample until we find an acceptable sample
+        while running:
+            accepted_vals = None
+            while accepted_vals is None:
+                # Get points from prior
+                xs = self._sampler.ask(self._n_workers)
```

Comment:
So for right now, I have added a separate variable which is set to 1 if we are not using parallelisation or set to `self._n_workers` if we are. I am not sure it would be useful for us to specify this variable in the constructor, but if you think that is the case, do let me know and I will make it so that this variable can be set in the constructor.